### ISSUE TITLE
Flow sorting issue for typing

### ISSUE THREAD

**Issue by k2snowman69** [snowcoders/sortier#18]
Input:
```
export type Data =
| $ReadOnly<{| +latter: string, +former: boolean|}>
| $ReadOnly<{| +differ: string, +former: boolean|}>
| $ReadOnly<{| +alternative: string, +former: boolean|}>;
```

Expected output:
```
export type Data =
| $ReadOnly<{| +alternative: string, +former: boolean|}>
| $ReadOnly<{| +differ: string, +former: boolean|}>
| $ReadOnly<{| +former: boolean, +latter: string|}>;
```

**Comment by k2snowman69** [comment]
Managed to get the base case working... now just working on nested structures

**Comment by k2snowman69** [comment]
Resolved with 4a077151bb76c3b1303cf0f9063460fc490acae0
